kit/src/components/embeds/file_embed/mod.rs

```diff
@@ -184,23 +201,30 @@ pub fn FileEmbed<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
                         aria_label: "file-icon",
                         if has_thumbnail {
                             rsx!(
-                                fullscreen_preview.then(|| rsx!(
-                                    Modal {
-                                        open: *fullscreen_preview.clone(),
-                                        onclose: move |_| fullscreen_preview.set(false),
-                                        transparent: false,
-                                        close_on_click_inside_modal: true,
-                                        dont_pad: true,
-                                        img {
-                                            id: "image-preview-modal-file-embed",
-                                            aria_label: "image-preview-modal-file-embed",
-                                            src: "{large_thumbnail}",
-                                            max_height: "80vh",
-                                            max_width: "80vw",
-                                            onclick: move |e| e.stop_propagation(),
-                                        },
+                                fullscreen_preview.then(|| {
+                                    if !temp_dir.exists() {
+                                        cx.props.on_press.call(Some(temp_dir.clone()));
                                     }
-                                )),
+                                    let temp_file_path_as_string = get_fixed_path_to_load_local_file(temp_dir.clone());
+                                    rsx!(
+                                            Modal {
+                                                open: *fullscreen_preview.clone(),
+                                                onclose: move |_| fullscreen_preview.set(false),
+                                                transparent: false,
+                                                close_on_click_inside_modal: true,
+                                                dont_pad: true,
+                                                img {
+                                                    id: "image-preview-modal-file-embed",
+                                                    aria_label: "image-preview-modal-file-embed",
+                                                    src: format_args!("{}", if temp_dir.exists()
+                                                        { temp_file_path_as_string}
+                                                        else {large_thumbnail} ),
```

Comment:
You may be able to avoid using `format_args` (or `format`) if both variables are strings.